client/rpmtrans.c

```diff
@@ -576,26 +593,23 @@ doCheck(PTDNFRPMTS pTS)
                     pr_crit("%s\n", msg);
                     if (rpmProblemGetType(prob) == RPMPROB_REQUIRES)
                     {
+                        char *pErrorStr = NULL;
                         uint32_t dwError = 0;
+
                         dwError = TDNFAllocateString(rpmProblemGetStr(prob), &pErrorStr);
-                        BAIL_ON_TDNF_ERROR(dwError);
+                        if (!dwError) {
+                            dwError = TDNFDetectPreTransFailure(pTS->pTS, pErrorStr);
+                        }
 
```

Comment:
Did you make this change to avoid calling `reportProblems()` if this fails?

Reply:
With this function now just reporting problems and no longer being responsible for validation/checking, it seems to me that it could now be "best effort", i.e., report on as many problems as it can manage to format instead of bailing early if something, say, goes wrong in `TDNFDetectPreTransFailure`.

What's your thinking?

Reply:
I think this needs reverting - use `BAIL_ON_TDNF_ERROR(dwError);` and remove the `if`, as it was before. We can later address calling `reportProblems()` when it's not an RPM error.

Reply:
Okay, will do. I will try to fix the free/cleanup logic on the bail case, then, to make sure the iterator et al. aren't potentially leaked.

Reply:
I've pushed an update that reverts this change and moves around the cleanup logic in that function to address the leak I mention above. Thanks for the feedback.